src/NuGet.Clients/NuGet.PackageManagement.UI/Models/DetailedPackageMetadata.cs

```diff
@@ -32,11 +32,19 @@ public DetailedPackageMetadata(IPackageSearchMetadata serverData, PackageDepreca
             Tags = serverData.Tags;
             DownloadCount = downloadCount;
             Published = serverData.Published;
-            DependencySets = serverData.DependencySets?
-                .Select(e => new PackageDependencySetMetadata(e))
-                ?? new PackageDependencySetMetadata[] { };
-            HasDependencies = DependencySets.Any(
-                dependencySet => dependencySet.Dependencies != null && dependencySet.Dependencies.Count > 0);
+
+            IEnumerable<PackageDependencyGroup> dependencySets = serverData.DependencySets;
+            if (dependencySets != null && dependencySets.Any())
+            {
+                DependencySets = dependencySets.OrderBy((ds) => ds.TargetFramework.DotNetFrameworkName)
+                    .OrderByDescending((ds) => ds.TargetFramework.Version)
+                    .Select(e => new PackageDependencySetMetadata(e));
```

Comment:
Can `DependencySets` be read multiple times?  If yes, I think you should add `.ToArray()` for reasons similar to this [statement](https://github.com/NuGet/NuGet.Client/pull/3229/files#diff-f6fe35ae56cabf8941953d92b67006c5R26-R29).

Reply:
next delta will have this.